assets/src/edit-story/components/canvas/gridview/index.js

```diff
@@ -161,33 +186,62 @@ ThumbnailSizeControl.propTypes = {
 function GridView() {
   const {
     state: { pages, currentPageIndex },
+    actions: { setCurrentPage, arrangePage },
   } = useStory();
   const [zoomLevel, setZoomLevel] = useState(2);
 
+  const width = zoomLevel * PREVIEW_WIDTH + THUMB_FRAME_WIDTH;
+  const height = zoomLevel * PREVIEW_HEIGHT + THUMB_FRAME_HEIGHT;
+
   return (
     <>
       <ThumbnailSizeControl value={zoomLevel} onChange={setZoomLevel} />
-      <Wrapper scale={zoomLevel}>
+      <Wrapper
+        scale={zoomLevel}
+        aria-label={__('Pages List', 'web-stories')}
+        onPositionChange={(oldPos, newPos) => {
+          const pageId = pages[oldPos].id;
+          arrangePage({ pageId, position: newPos });
+          setCurrentPage({ pageId });
+        }}
+      >
         {pages.map((page, index) => {
           const isCurrentPage = index === currentPageIndex;
 
           return (
-            <DraggablePage
-              key={index}
-              ariaLabel={
-                isCurrentPage
-                  ? sprintf(
-                      __('Page %s (current page)', 'web-stories'),
-                      index + 1
-                    )
-                  : sprintf(__('Page %s', 'web-stories'), index + 1)
-              }
-              isActive={isCurrentPage}
-              pageIndex={index}
-              width={zoomLevel * PREVIEW_WIDTH + THUMB_FRAME_WIDTH}
-              height={zoomLevel * PREVIEW_HEIGHT + THUMB_FRAME_HEIGHT}
-              dragIndicatorOffset={GRID_GAP / 2}
-            />
+            <PagePreviewContainer key={`page-${index}`}>
+              {index === 0 && (
```

Comment:
I don't quite understand why separators need to be inside the PagePreviewContainer. Could you pls clarify?

Reply:
Because otherwise they'd be counted as a grid item and take up as much space as a page preview, so unless we change the css grid, this has to be one element

Reply:
Got it.